### ISSUE TITLE
WebSearch: 'e' method in search_unit()

### ISSUE THREAD

**Issue by kaplun** [inveniosoftware/invenio#3095]
Currently when one uses `search_unit(p=... f=..., m='e')` the current code (in legacy):

``` python
    elif m == 'a' or m == 'r' or f == 'subject':
        # we are doing either phrase search or regexp search
    [...]
    else:
        # we are doing bibwords search by default
        hitset = search_unit_in_bibwords(p, f, wl=wl)
```

makes search_unit() to use `search_unit_in_bibwords()` (i.e. the WORD index), instead of `search_unit_in_idxphrases()` as intended.

Is this intended or a bug?

**Comment by kaplun** [comment]
@tiborsimko do you know why?

**Comment by kaplun** [comment]
This currently is causing misbehavior in INSPIRE for the citation linking.

**Comment by jalavik** [comment]
The `subject` part was an addition made for INSPIRE back in the day, but I guess you refer to the `m='e'` part.

**Comment by tiborsimko** [comment]
**1.** Regarding "subject", we have talked about it independently with @jalavik and @Kennethhole.  This was added via INSPIRE search branch, but IMO it should be removed.  If a special treatment for "subject" is needed, then it is better achieved via dedicated tokeniser, like for author field.  Actually, @Kennethhole has a client that uses an index called "subject" with another default tokeniser...  So this creates a problem. `search_unit()` should not have hard-coded index names, we are actually going to remove it for v1.2.1.

**2.** Regarding "e" matching type, the situation is kind of subtle.  Users are usually coming via high-level search APIs such as "search_pattern()" etc.  These call tokeniser "create_basic_search_units()" where matching type "e" can be used, signifying "exact match".  When you look at the code of "create_basic_search_units()", when "e" is passed on the input, the function changes the unit to "a" (=access).  Hence "search_unit()" later sees only "a" usually; it never sees "e".
Hence we should not really need to catch it there.

**TL;DR** The matching type "e" is used in higher-level search APIs, and is transformed to "a" before the lower-level APIs like "search_unit()" are called.  (The same happens for "p" matching type, for example.) 

Hence question: in which usage scenario do you see "search_unit(p=..., f=..., m='e')" being called? This could be an issue of changing the client call... in which case we don't need to change anything in the search code, but just adapt the client.  

Basically, we have two options: either (1) simply better document which "m" values are expected at each search level step; or (2) enrich search_unit() to accept all the other matching types too, i.e. not only "e" but also "p" and friends.
